addons/web/static/tests/helpers/mock_server.js

```diff
@@ -1521,10 +1537,18 @@ var MockServer = Class.extend({
                 } else {
                     res[groupByField] = val;
                 }
-
+                
+                if (['date', 'datetime'].includes(field.type)) {
+                    if (typeof res.__range === 'undefined') {
```

Comment:
Strange wording ? Isn't it a bit low-level ? (to check if something like `if (!res.__range) { blah }` could work ?)